### PR TITLE
zenodo and authors: Upgraded handling of the authors information.

### PR DESCRIPTION
**Pull request by rbeyer**
## Description
- Upgraded zenodo_to_authors.py to write out the AUTHORS.rst file in alphabetical order, and added affiliations, if any.
- Created zenodo_order.py to re-arrange the "creator" list in the .zenodo.json file so that the designated first author (project lead) comes first followed by everyone else in alphabetical order.
- Committed the new versions of .zenodo.json and AUTHORS.rst
- Added my own name, affiliation, and ORCID iD

## Related Issue
#3810
And also discussed in ISIS TC meeting.

## Motivation and Context
This addresses the alphabetical ordering of authors in the generated AUTHORS.rst file.  The new script also sets the ordering in the .zenodo.json file.

## How Has This Been Tested?
Ran it locally, did some trouble-shooting.  Seems to work.

## Types of changes
- Documentation change (update to the documentation; no code change)

## Checklist:
- I **could not possibly have read** and agreed to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/) since it is still 404ing.
- I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/scripts/zenodo_order.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import argparse
+import json
+import sys
+
+from operator import itemgetter
+
+# This is nominally the ISIS project lead or whomever should be
+# the first author in the Zenodo authors listing.
+first_author = "Laura, Jason"
```
Comment (jessemapel) [inline isis/scripts/zenodo_order.py]: This should be after the license comment.

I also feel like this is an argument, but not one I want to mess with often. Making it something with a default value is kind of odd, though. Is there a better way to handle this?

File: isis/scripts/zenodo_order.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    sys.exit(
+        f"The designated first author ({project_lead_name}) was not found "
+        f"in {args.zenodo_json}"
+    )
+
+first = creators_list.pop(first_idx)
+
+# Create new creators list, put lead first, and then alphabetize
+new_creators = [first]
+new_creators += sorted(creators_list, key=itemgetter("name"))
```
Comment (jessemapel) [inline isis/scripts/zenodo_order.py]: This is not very pythonic, list comprehension may be better suited here. Something like:

```
new_creators = [creator for creator in parsed_json["creators"] if creator["name"] == first_author]
if not new_creators:
    sys.exit(
        f"The designated first author ({project_lead_name}) was not found "
        f"in {args.zenodo_json}"
    )
new_creators += sorted([creator for creator in parsed_json["creators"] if creator["name"] != first_author], key=itemgetter("name"))
```

File: isis/scripts/zenodo_order.py (hunk 1 above)
Comment (rbeyer) [reply isis/scripts/zenodo_order.py]: If we knew more about what keys were allowed in the .zenodo.json file (see previous discussions on the woeful lack of documentation), there might be a way to define a key/value pair in one of the creator records to designate the project lead, and then we wouldn't have to force it here.

Aside from that, sure making it a defaulted argument is fine, too.  I'll work it.

File: isis/scripts/zenodo_order.py (hunk 2 above)
Comment (rbeyer) [reply isis/scripts/zenodo_order.py]: That's a fine color for the bike shed.  Sure, I'll make this change, too.

Comment (rbeyer) [reply isis/scripts/zenodo_order.py]: I decided I didn't like looping through the creators list twice, and I also didn't like the long list comprehension (seemed hard to read to me).  I did make a change, just not the one you suggested.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**